misk/src/main/kotlin/misk/client/PeerClientFactory.kt

```diff
@@ -0,0 +1,113 @@
+package misk.client
+
+import com.google.common.cache.CacheBuilder
+import com.google.common.cache.CacheLoader
+import com.google.inject.Provides
+import misk.clustering.Cluster
+import misk.config.AppName
+import misk.inject.KAbstractModule
+import misk.security.cert.X500Name
+import misk.web.WebConfig
+import misk.web.jetty.JettyService
+import okhttp3.OkHttpClient
+import java.security.cert.X509Certificate
+import java.util.concurrent.TimeUnit
+import javax.inject.Singleton
+
+/**
+ * Binds a [PeerClientFactory] that calls peers on the HTTPS port of this process's server,
+ * as determined by the SSL port in the [WebConfig].
+ */
+class PeerClientModule : KAbstractModule() {
+  @Provides @Singleton fun peerClientFactory(
+    @AppName appName: String,
+    httpClientsConfig: HttpClientsConfig,
+    httpClientFactory: HttpClientFactory,
+    webConfig: WebConfig
+  ): PeerClientFactory {
+    check(webConfig.ssl?.port ?: 0 > 0) { "server must have static HTTPS port" }
+
+    return PeerClientFactory(
+        appName = appName,
+        httpClientsConfig = httpClientsConfig,
+        httpClientFactory = httpClientFactory,
+        httpsPort = webConfig.ssl!!.port
+    )
+  }
+}
+
+/**
+ * For testing.
+ *
+ * Binds a [PeerClientFactory] that calls peers on the HTTPS port of this process's server,
+ * as determined by the Jetty server's port.
+ */
+class JettyPortPeerClientModule : KAbstractModule() {
+  @Provides @Singleton fun peerClientFactory(
+    @AppName appName: String,
+    httpClientsConfig: HttpClientsConfig,
+    httpClientFactory: HttpClientFactory,
+    jetty: JettyService
```

Comment:
I wonder if we want to have a `Provider<ServerPorts>` that returns a `data class ServerPorts(val http: Int, val https: Int)` object with the actual HTTP/HTTPS ports, derived from the config in normal mode and the `JettyService` for dynamic port assignmint in tests